receiver.py

```diff
@@ -63,16 +70,58 @@ def async_tx(self, command):
         """
         self.outbound.append(command)
 
-    def __enter__(self):
-        #Change this to search for USB, Unhardcode the ports
-        if _platform == "linux" or _platform == "linux2":
-            self.ser = serial.Serial('/dev/ttyUSB0', 115200)
-        elif _platform == "win32":
-            self.ser = serial.Serial('COM5', 115200)
-        self.xbee = ZigBee(self.ser)
-        print 'xbee created/initialized'
+
+    def reconnect_xbee(self):
+
+        #detect platform and format port names
+        if _platform.startswith('win'):
+            ports = ['COM%s' % (i + 1) for i in range(256)]
+        elif _platform.startswith('linux'):
+            # this excludes your current terminal "/dev/tty"
+            ports = glob.glob('/dev/tty[A-Za-z]*')
```

Comment:
Also, this is overly generous glob. Plugging in our xbee -> usb/serial adapter will always put it as a usb tty, so you can make the glob narrower. (ie. `glob('/dev/ttyUSB*')`).